src/main/kotlin/com/replaymod/gradle/preprocess/PreprocessTask.kt

```diff
@@ -246,15 +251,35 @@ class CommentPreprocessor(private val vars: Map<String, Int>) {
             n++
             val trimmed = line.trim()
             val mapped = if (trimmed.startsWith(kws.`if`)) {
-                val result = trimmed.substring(kws.`if`.length).trim().evalExpr()
-                ifStack.push(result)
+                trueConditionFound = trimmed.substring(kws.`if`.length).trim().evalExpr()
+                elseFound = false
+                ifStack.push(trueConditionFound)
                 indentStack.push(line.indentation)
-                active = active && result
+                active = active && trueConditionFound
+                line
+            } else if (trimmed.startsWith(kws.elseif)) {
+                if (ifStack.isEmpty()) {
+                    throw ParserException("Unexpected elseif in line $n of $fileName")
+                }
+                if (elseFound) {
+                    throw ParserException("Unexpected elseif after else in line $n of $fileName")
+                }
+                if (trueConditionFound) {
+                    active = false
```

Comment:
This should be set the same way as in the `else` case, in particular, the boolean on top of the if stack needs to be flipped, otherwise it'll cause issues when nested.
The `indentStack` should also be updated (you can just move the last three lines out of the following `!trueConditionFound` block so they're applicable to both).

Reply:
You seem to have missed this one. It's the cause for my second example, which I believe would still fail. Looks good otherwise.

Reply:
It was an intentional miss, because I didn't quite understand it. I did test the second example and it parses correctly.